### PR TITLE
Refactor: tests to a separated layout, wiki package in src/

### PR DESCRIPTION
**Pull request by benjaoming**
Refactor:

 * All test code moved to `tests/`
 * All of the `wiki` package now lives in `src/`
 * Redid some pep8 because some of the codebase was unlinted

From Pytest's *Good Integration Practices*:

> This layout prevents a lot of common pitfalls and has many benefits, which are better explained in [this excellent blog post by Ionel Cristian Mărieș](https://blog.ionelmc.ro/2014/05/25/python-packaging).

### CODE REVIEW

**Comment by benjaoming** [comment]
This already caught one error, having had test+development artifacts in the dist tarball/whl on Pypi.

It's definitely going to give a much better guarantee that gaps between local development environments and distributed environments don't get confused in the tests: We test the dist, not the dev environment, as the blog post "Packaging a python library" says.

File: src/wiki/plugins/haystack/__init__.py [hunk 1]
```diff
@@ -64,7 +64,6 @@ def __call__(self, request):
                 models.URLPath.root().article,
                 self.request.user):
             self.results = self.__filter_can_read(self.request.user)
-            #self.results = self.results.filter(current_revision__deleted=False)
```
Comment (benwah) [inline src/wiki/plugins/haystack/__init__.py]: giphy

File: src/wiki/plugins/images/templatetags/wiki_thumbnails.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+from __future__ import absolute_import
```
Comment (benwah) [inline src/wiki/plugins/images/templatetags/wiki_thumbnails.py]: Why is this file needed?

File: tests/core/test_basic.py [hunk 3]
```diff
@@ -7,8 +7,8 @@
 from wiki.forms import Group
 from wiki.models import URLPath
 
-from .base import wiki_override_settings
-from .testdata.models import CustomGroup
+from ..base import wiki_override_settings
```
Comment (benwah) [inline tests/core/test_basic.py]: 👍

File: src/wiki/plugins/images/templatetags/wiki_thumbnails.py (hunk 2 above)
Comment (benjaoming) [inline src/wiki/plugins/images/templatetags/wiki_thumbnails.py]: As of now, it's needed because a lot of HTML files say `{% load wiki_thumbnails %}` but I don't know if adding the `register` from sorl actually makes any difference.. I should try deleting it, you're right!

Comment (benjaoming) [inline src/wiki/plugins/images/templatetags/wiki_thumbnails.py]: Just verified it, this *is* actually the reason :) I think it's to provide a decent wrapper for sorl.thumbnail on case it changes API or we add an alternative. Or maybe it's there for some historic reason.

**Comment by benjaoming** [comment]
Gonna get this merged now and not let it sit in the way of future PRs!
